File: Source/Fantasy.Jobs/Resources/IResourceRequestQueue.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Fantasy.Jobs.Resources
{
    public interface IResourceRequestQueue
    {
        void RegisterResourceRequest(Guid jobId, ResourceParameter[] parameters);
        void UnregisterResourceRequest(Guid jobId);

        ResourceParameter[] GetRequiredResources(Guid jobId);
    }
}

File: branches/Stheno/Fantasy.Jobs/src/fantasy/jobs/resources/IResourceRequestQueue.java
package fantasy.jobs.resources;

import java.util.UUID;

public interface IResourceRequestQueue
{
	void registerResourceRequest(UUID jobId, ResourceParameter[] parameters) throws Exception;
	void unregisterResourceRequest(UUID jobId) throws Exception;

	ResourceParameter[] getRequiredResources(UUID jobId) throws Exception;
}